File: lcm/lcm.cpp
#include <iostream>

using namespace std;

int gcd(int a, int b)
{
    while(b){
        b^=a^=b^=a%= b;
    }
    return a;
}

int lcm(int a, int b)
{
    return a / gcd(a, b) * b;
}

int main()
{
    cout << lcm(8, 12) << endl;
    return 0;
}

File: lcm/lcm.go
package main

import "fmt"

func gcd(a, b int) int {
	for b != 0 {
		a %= b
		b ^= a
		a ^= b
		b ^= a
	}
	return a
}

func lcm(a, b int) int {
	return a / gcd(a, b) * b
}

func main() {
	fmt.Println(lcm(8, 12))
}
